flexget/plugins/input/anidb_list.py

```diff
@@ -26,69 +28,100 @@ class AnidbList(object):
-                if not a:
-                    log.debug('No title link found for the row, skipping')
-                    continue
-
-                anime_title = a.string
-                if config.get('strip_dates'):
-                    # Remove year from end of series name if present
-                    anime_title = re.sub(r'\s+\(\d{4}\)$', '', anime_title)
-
-                link = ('http://anidb.net/perl-bin/' + a.get('href'))
-
-                anime_id = ""
-                match = re.search(r'aid=([\d]{1,5})', a.get('href'))
-                if match:
-                    anime_id = match.group(1)
-
-                entry = Entry()
-                entry['title'] = anime_title
-                entry['url'] = link
-                entry['anidb_id'] = anime_id
-                entry['anidb_name'] = entry['title']
-                entries.append(entry)
-            else:
-                log.verbose('Entry does not match the requested type')
+        elif config['type'] == 'ovas':
+            entry_type = 'Type: OVA'
+        while True:
+            soup = get_soup(page.text)
+            soup_table = soup.find('table', class_='wishlist').find('tbody')
+
+            trs = soup_table.find_all('tr')
+            if not trs:
+                log.verbose('No movies were found in AniDB list: mywishlist')
+                return entries
+            for tr in trs:
+                if tr.find('span', title=entry_type):
+                    a = tr.find('td', class_='name').find('a')
+                    if not a:
+                        log.debug('No title link found for the row, skipping')
+                        continue
+
+                    anime_title = a.string
+                    if config.get('strip_dates'):
+                        # Remove year from end of series name if present
+                        anime_title = re.sub(r'\s+\(\d{4}\)$', '', anime_title)
+
+                    entry = Entry()
+                    entry['title'] = anime_title
+                    entry['url'] = (self.anidb_url + a.get('href'))
+                    entry['anidb_id'] = tr['id'][1:]  # The <tr> tag's id is "aN..." where "N..." is the anime id
+                    log.debug('%s id is %s', entry['title'], entry['anidb_id'])
+                    entry['anidb_name'] = entry['title']
+                    entries.append(entry)
+                else:
+                    log.verbose('Entry does not match the requested type')
+            try:
+                # Try to get the link to the next page.
+                next_link = soup.find('li', class_='next').find('a')['href']
+            except TypeError:
+                # If it isn't there, there are no more pages to be crawled.
+                log.verbose('No more pages on the wishlist.')
+                break
+            comp_link = self.anidb_url + next_link
+            log.debug('Requesting: %s', comp_link)
+            page = task.requests.get(comp_link, headers=task_header)
```

Comment:
Needs a try-except as well.

Reply:
I wrapped this in a try:except, but I didn't raise a plugin error from it, as it is at the end of the loop where we have at least the one page of items on the list as not to throw those out. Instead it warns the user that there was an issue and logs the error that was raised.